File: AddressSearchSample/Objective-C/AddressSearchSample/AddressSearchAttributesViewController.m
//
//  AddressSearchAttributesViewController.m
//  AddressSearchSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

#import "AddressSearchAttributesViewController.h"

@implementation AddressSearchAttributesViewController

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"resultSegue"]) {
        AddressSearchResultViewController *addressSearchResultViewController = segue.destinationViewController;
        NTAddressSearchParameter *param = [[NTAddressSearchParameter alloc] init];
        param.houseNo = _txtHouseNo.text;
        param.moo = _txtMoo.text;
        param.soi = _txtSoi.text;
        param.road = _txtSoi.text;
        param.adminLevel1 = _txtAdminLevel1.text;
        param.adminLevel2 = _txtAdminLevel2.text;
        param.adminLevel3 = _txtAdminLevel3.text;
        param.postcode = _txtPostcode.text;
        
        addressSearchResultViewController.param = param;
    }
}

@end

File: AddressSearchSample/Swift/AddressSearchSample/AddressSearchAttributesViewController.swift
//
//  AddressSearchAttributesViewController.swift
//  AddressSearchSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

import UIKit
import NOSTRASDK

class AddressSearchAttributesViewController: UIViewController {

    @IBOutlet weak var txtHouseNo: UITextField!
    @IBOutlet weak var txtMoo: UITextField!
    @IBOutlet weak var txtSoi: UITextField!
    @IBOutlet weak var txtRoad: UITextField!
    @IBOutlet weak var txtAdminLevel1: UITextField!
    @IBOutlet weak var txtAdminLevel2: UITextField!
    @IBOutlet weak var txtAdminLevel3: UITextField!
    @IBOutlet weak var txtPostcode: UITextField!
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "resultSegue" {
            let resultViewController = segue.destination as? AddressSearchResultViewController;
            
            let param = NTAddressSearchParameter();
            
            param.houseNo = txtHouseNo.text;
            param.moo = txtMoo.text;
            param.soi = txtSoi.text;
            param.road = txtRoad.text;
            param.adminLevel1 = txtAdminLevel1.text;
            param.adminLevel2 = txtAdminLevel2.text;
            param.adminLevel3 = txtAdminLevel3.text;
            param.postcode = txtPostcode.text;
            
            resultViewController?.addressSearchParam = param;
            
        }
    }

}
